src/main.c

```diff
@@ -248,7 +248,6 @@ int parse_ethhdr(unsigned char* buffer, int size) {
 }
 
 int read_args(int argc, char *argv[]) {
-	m.allow_gateway_s = NULL; // because C likes to be undefined and shoot itself
```

Comment:
This one is VERY intentional. I'm setting the pointer to NULL since there's a check later to see if this string has been set. Since C compilers don't guarantee default values of NULL or 0, I'd suggest this one stays on.

Otherwise have fun debugging string issues that are OS/runtime/compiler related.

Reply:
Also shouldn't your compiler complain about "condition depends on uninitialized value"

Reply:
I didn't, so I was not sure. I would add another flag to check this.

By the way, dut during debugging I found out that this does clear the load_config value. ;)

We need kind of a constructor to init all our fields to null after I do enable the unexpected behaivfor flag `-fsanitize=undefined`

All the memory leek fixes are kind of pointless to fix, because they are all at the one time executed init. But we need to fix them to be able to let the tool running in debug mode to find importand issues.